contrib/99designs/gqlgen/tracer.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package gqlgen contains an implementation of a gqlgen tracer, and functions to construct and configure the tracer.
+// The tracer can be passed to the gqlgen handler (see package github.com/99designs/gqlgen/handler)
+package gqlgen
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/vektah/gqlparser/v2/ast"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+const (
```

Comment:
Question: Is there a top level span which wraps all graphql spans or are the spans below sibling spans? 

Here's the structure of a graphql trace in [python](https://github.com/DataDog/dd-trace-py/blob/bae194fd870511ab6edc3f4efa3552668aa7264a/tests/snapshots/tests.contrib.graphql.test_graphql.test_graphql.json#L3) and in [.net](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/test/snapshots/GraphQL4Tests.SubmitsTraces.verified.txt#L5). 

The top level span is `graphql.request` in the python tracer but this span does not exist in the node or .net implementations.

Reply:
Yep the top level span is `graphql.query` and the resource name of that span is the raw query. There are then three children spans that are children for parse, validate, and execute.
(From the internal thoughts, I thought Python's top level node was called `graphql.query`, not `graphql.request`?)

Reply:
It was ‘graphql.query’ however this was changed recently because we thought it could be misleading (if the request was a mutation or a subscription we still set that span name). We opted to use ‘graphql.request’ instead since that’s a convention used in most of our web integrations

Reply:
Talked offline. Changed the default to be "graphql.request". However, this Go implementation is nice in that we can see the operation type (e.g. "subscription", "mutation", "query") so we can use that in the operation name to be more specific, which is what is happening now. I don't think graphql.request will ever be needed as the default, but it's there just in case.